examples/rest/calendar_recurrence.py

```diff
@@ -131,6 +134,13 @@ def to_internal_value(self, data: dict):
             dates=data.pop("dates", None),
             weekdays=list(data.pop("weekdays", set())),
         )
+        if data["pattern"] != CalendarRecurrence.Pattern.Weekly.value:
+            data.pop("weekdays", None)
+        if data["pattern"] != CalendarRecurrence.Pattern.Monthly.value:
+            data.pop("days", None)
+        if data["pattern"] != CalendarRecurrence.Pattern.Yearly.value:
+            data.pop("dates", None)
+
```

Comment:
FYI: this same code is on the other project. Probably a new task should be opened to check the functionality there.